File: cpp/prime.cpp
#include<stdio.h>
#include<stdlib.h>
#include<time.h>

bool isPrime(int number)
{
	for(int n = 2; n < number / 2; n++)
	{
		if(number % n == 0) return false;
	}
	return true;
}

int main(int argc, char *args[])
{
	int limit = 0;
	if(argc == 2) limit = atoi(args[1]);
	if(limit < 1) limit = 10000;

	clock_t start = clock();
	int count = 0;
	for(int number = 2; ; number++)
	{
		if(!isPrime(number)) continue;
		count++;
		if(count == limit)
		{
			printf("%dth prime is %d\n", limit, number);
			break;
		}
	}
	printf("[cpp] duration is %f sec\n", (double)(clock() - start)/CLOCKS_PER_SEC);
}

File: ruby/prime.rb

def is_prime(num)
	for n in 2...(num/2+1)
		if num % n == 0 then
			return false
		end
	end
	return true
end

def search()
	limit = 10000
	if ARGV.count == 1 then
		limit = ARGV[0].to_i
	end
	start = Time.now
	count = 0
	num = 1
	while count < limit
		num += 1
		if is_prime(num) then
			count += 1
		end
	end
	puts "%dth prime is %d" % [limit, num]
	puts "[ruby] duration is #{Time.now - start} sec"
end

search()
